### ISSUE TITLE
[unused variables] Report unused use() variable

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#188]
**See next example:**

```php
$a = null;
$b = function () use ($a) {
};
```

Note that `$b` uses `$a` only "virtually", but it is never readed inside it. I guess that in this case it should be represented like a `unused referenced variable`.

**Comment by kalessil** [comment]
That's sweet, a good case =)

**Comment by kalessil** [comment]
Added, reports unsed use-variables and reports only-write cases for the variables

**Comment by rentalhost** [comment]
@kalessil No binary? 

Did you checked if it will warn too:

```php
$b = function () use (&$a) {
};
```

**Comment by kalessil** [comment]
Fresh binary pushed now. And pretty similar cases is tested also.

**Comment by rentalhost** [comment]
@kalessil I tried last binary, but unfortunetelly the `&$a` doesn't is reported. 😢

**Comment by rentalhost** [comment]
Off-topic: there some doc that I can read to try to help with some easy-picks PRs? I am not a Java programmer, but I think that I can help in some cases, but I don't know from where I should starts.

**Comment by kalessil** [comment]
Right, added the test case; fix will be available today.

Yep, docs are available, you can find  it [here](https://github.com/kalessil/phpinspectionsea/blob/master/docs/DEVELOPERS.md).

**Comment by kalessil** [comment]
I'ts pretty easy to add tests, we have lists of missing tests [here](https://github.com/kalessil/phpinspectionsea/blob/master/RULES.md). Some text samples even [available](https://github.com/kalessil/phpinspectionsea/tree/master/php).

**Comment by rentalhost** [comment]
@kalessil  Can you share your Java code style settings?

**Comment by kalessil** [comment]
I use defaults, except aligning sequential assignments manually.

**Comment by rentalhost** [comment]
Auto alignments doesn't works?

**Comment by kalessil** [comment]
I didn;t find where I can switch this on...

**Comment by rentalhost** [comment]
Seems that the align could be done by enable the "group declarations" on Java code style / Wrapping and Braces.

---

Okay, I have done the initial setup and I can run the tests, but I received 13 failed tests. I think that we can have some problem here, because in some comment you told that all is green. Or maybe I'm just forgetting something.

**For instance:**

> com.intellij.rt.execution.junit.FileComparisonFailure: usage-of-silence-operator.php: Missing (4:5/1) :'@' (Try to avoid using the @, as it hides problems and complicates troubleshooting.)

**Differences:**

```diff
  namespace {
-     <warning descr="Try to avoid using the @, as it hides problems and complicates troubleshooting.">@</warning>touch('...');
+     @touch('...');
      echo <warning descr="Try to avoid using the @, as it hides problems and complicates troubleshooting.">@</warning>$undefinedVar;
  }
```

I'm running on `master` and I have attached the problems here.

[Test Results - Tests.zip](https://github.com/kalessil/phpinspectionsea/files/902493/Test.Results.-.Tests.zip)

**Comment by kalessil** [comment]
Pull new changes from the master - I pushed fixed ~10 mins ago. Will check the option today, thanks!

**Comment by rentalhost** [comment]
Yep! All green now. Great! Now I should try some easy-picks 😋
